scipy/linalg/_decomp_polar.py

```diff
@@ -99,6 +99,18 @@ def polar(a, side="right"):
     a = np.asarray(a)
     if a.ndim != 2:
         raise ValueError("`a` must be a 2-D array.")
+    if a.size == 0:
+        dt = a.dtype
```

Comment:
This is not right: the output dtypes should match the those for the non-empty case

```
In [4]: polar(np.zeros((0, 3), dtype=int))
Out[4]: 
(array([], shape=(0, 3), dtype=int64),
 array([[94045842768320,              0,              0],
        [             0,              0,              0],
        [             0,              0,              0]]))
```